File: 백준/Silver/10816. 숫자 카드 2/숫자 카드 2.py
hashmap = {}

n = int(input()) # 10
N = input().split() # 6 3 2 10 10...

m = int(input()) # 8 
M = input().split() # 10 9 -5 2 3 4 5 ...

for num in N:
    if num in hashmap:
        hashmap[num] += 1
    else:
        hashmap[num] = 1
        
for num in M:
    if num in hashmap:
        print(hashmap[num], end=' ')
    else:
        print(0, end= ' ')

File: 백준/Silver/10816. 숫자 카드 2/숫자 카드 2.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st;

        int N = Integer.parseInt(br.readLine());
        HashMap<Integer, Integer> map = new HashMap<Integer, Integer>();

        st = new StringTokenizer(br.readLine());

        // hashmap에 존재하지 않으면 value 1로 지정, 존재하면 value++
        for(int i = 0; i < N; i++){
            int num = Integer.parseInt(st.nextToken());
            if(!map.containsKey(num)){
                map.put(num, 1);
            }
            else {
                map.put(num, map.get(num) + 1);
            }
        }

        int M = Integer.parseInt(br.readLine());
        StringBuilder sb = new StringBuilder();
        st = new StringTokenizer(br.readLine());

        // hashmap에 존재하면 value 불러오기, 존재하지 않으면 0
        for(int i = 0; i < M; i++){
            int num = Integer.parseInt(st.nextToken());
            if(map.containsKey(num)){
                sb.append(map.get(num)).append(" ");
            }
            else {
                sb.append(0).append(" ");
            }
        }

        System.out.println(sb);
    }
}